src/http.rs

```diff
@@ -88,53 +89,47 @@ where
                 .insert("User-Agent", self.user_agent.clone());
         }
 
-        debug!(
-            message = "Sending HTTP request.",
-            uri = %request.uri(),
-            method = %request.method(),
-            version = ?request.version(),
-            headers = ?request.headers(),
-            body = %FormatBody(request.body()),
-        );
+        emit!(http_client::AboutToSendHTTPRequest { request: &request });
```

Comment:
Should this be called in `async move` body or doesn't matter? (i.e. when future will be polled)

Reply:
Good point!

Reply:
Ah, actually, we're referencing the request there, and it's consumed after we prepare the request. I think preparing the request outside of the future is more important than doing the emit in the right spot, so I don't think it makes sense to move this.
The name could be adjusted to better relate to the point at which we call the emit (something more precise than `AboutToSendHTTPRequest`) - but I don't think it that bad.

Reply:
Yep, I agree with you.